docs/kind-3000-pillar-registration.md

```diff
@@ -72,27 +88,53 @@ These tags make the event discoverable by NIP-73 compatible clients. The `i` tag
 - `i`: `["i", "zenon:1:address:<pillar_bech32_address>"]`
 - `k`: `["k", "zenon:address"]`
 
-Example: `["i", "zenon:1:address:z1qqjnwjjpnue8xmmpanz6csze6tcmtzzdtfsww7"]`, `["k", "zenon:address"]`
+Example: `["i", "zenon:address:z1qqjnwjjpnue8xmmpanz6csze6tcmtzzdtfsww7"]`, `["k", "zenon:address"]`
 
 ### `alt`
 A human-readable summary of the event's purpose, as per [NIP-31](https://github.com/nostr-protocol/nips/blob/master/31.md). The `content` field **MUST** be empty.
 
 Example: `["alt", "Registration for PillarOne on HC1 Incubator"]`
 
+### `e` (previous event link)
+The id of the previous pillar registration event for the same pillar/address, used to form a verifiable chain of updates. Omit on the first registration.
+
+Example: `["e", "4376c65d2f232afbe9b882a35baa4f6fe8667c4e684749af565f981833ed6a65"]`
+
+## Canonical Signature Message
+
+To prevent replay across contexts and to ensure wallet compatibility, the external signature MUST be produced over the following canonical, single-line UTF-8 string constructed from the event fields and tags exactly as published, joined with the `|` character in this exact order:
+
+```
+hc1:registration|kind=3000|i=<value of i tag>|k=<value of k tag>|nostr_pubkey=<event.pubkey, lowercase hex>|sequence=<value of sequence tag>|created_at=<event.created_at>
```

Comment:
Apparently the previous signature message `pillar-registration:<pubkey>` was not good enough to bind the signature to the specific event and it was open to some attacks. Does this look more reasonable?